.circleci/config.yml

```diff
@@ -0,0 +1,59 @@
+version: 2.0
+jobs:
+    build:
+      docker:
+        - image: madmongo1/fedora-devel:latest
+
+      working_directory: /tmp
+
+      steps:
+        - run:
+            name: fix ssh keys
+            working_directory: ~
+            command: |
+              ssh-keyscan github.com >> githubKey
+              ssh-keygen -lf githubKey
+              cat githubKey >> ~/.ssh/known_hosts
+
+        - run:
+            name: clone boost
+            working_directory: /tmp
+            command: |
+              BOOST_BRANCH=master
+              if [ "${CIRCLE_BRANCH}" != "master" -a "${CIRCLE_BRANCH}" != "refs/heads/master" ]
+              then
+                BOOST_BRANCH=develop
+              fi
+              git clone -b $BOOST_BRANCH --depth 1 https://github.com/boostorg/boost.git
+
+        - run:
+            name: update submodules
+            working_directory: /tmp/boost
+            command: |
+              git submodule update --init --recursive --depth 1 --jobs 4
```

Comment:
A depth=1 submodule update may result in spurious failures whenever somebody commits to a dependency and the root repo isn't updated yet.

Reply:
confirmed. I'll use 10 as per the script